finagle-zipkin-core/src/main/scala/com/twitter/finagle/zipkin/core/RawZipkinTracer.scala

```diff
@@ -0,0 +1,149 @@
+package com.twitter.finagle.zipkin.core
+
+import com.twitter.conversions.time._
+import com.twitter.finagle.stats.StatsReceiver
+import com.twitter.finagle.thrift.thrift
+import com.twitter.finagle.tracing
+import com.twitter.finagle.tracing._
+import com.twitter.finagle.util.DefaultTimer
+import com.twitter.util._
+import java.net.InetSocketAddress
+import java.nio.ByteBuffer
+import scala.language.reflectiveCalls
+
+/**
+ * Receives the Finagle generated traces and sends them off to Zipkin via scribe.
+ * @param statsReceiver We generate stats to keep track of traces sent, failures and so on
+ * @param timer A Timer used for timing out spans in the [[DeadlineSpanMap]]
+ */
+abstract private[zipkin] class RawZipkinTracer(
+  statsReceiver: StatsReceiver,
+  timer: Timer = DefaultTimer.twitter
+) extends Tracer {
+  private[this] val ErrorAnnotation = "%s: %s" // annotation: errorMessage
+
+  // this sends off spans after the deadline is hit, no matter if it ended naturally or not.
+  private[this] val spanMap: DeadlineSpanMap =
+    new DeadlineSpanMap(sendSpans, 120.seconds, statsReceiver, timer)
+
+  protected[core] def flush() = spanMap.flush()
```

Comment:
since you are modifying this file, if you don't mind making the world ever so slightly better, can you annotate this with the return type?